addons/mail/models/mail_message.py

```diff
@@ -927,7 +927,8 @@ def _get_message_id(self, values):
     def _invalidate_documents(self):
         """ Invalidate the cache of the documents followed by ``self``. """
         for record in self:
-            if record.model and record.res_id and 'message_ids' in self.env[record.model]:
+            ir_model = self.env['ir.model'].search([('model', '=', record.model)], limit=1)
```

Comment:
I would use issubclass(type(record.model), self.pool['mail.thread']) to avoid a search here. This will avoid the access and query count errors.

Reply:
`record.model` is a string then using `type(record.model)` the result is `str`

Did you mean:
```bash
>>> issubclass(type(self.env['account.invoice']), self.pool['mail.thread'])
True
>>> issubclass(type(self.env['stock.move']), self.pool['mail.thread'])
False
```

FYI using `isinstance` you can avoid using `type`.

I mean,
```bash
>>> isinstance(self.env['account.invoice'], self.pool['mail.thread'])
True
>>> isinstance(self.env['stock.move'], self.pool['mail.thread'])
False
```

So What about using `isinstance(record.model, self.pool['mail.thread'])`?

Reply:
In fact, you can use just class vs class

```bash
>>> issubclass(self.pool['stock.move'], self.pool['mail.thread'])
False
>>> issubclass(self.pool['account.invoice'], self.pool['mail.thread'])
True
```

I mean, 
```diff
-            if record.model and record.res_id and 'message_ids' in self.env[record.model]:
+            if record.model and record.res_id and issubclass(self.pool[record.model], self.pool['mail.thread']):
```

Reply:
Yes indeed, thanks for the correction.